### PR TITLE
Add new cert verify status

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #1718

Commit 1： Add new status to distinguish verify cert chain status.

- When verify cert chain integrity failed, the status is VERIF_FAIL.
- When verify cert chain authority failed, the status is NO_AUTHORITY.

Commit 2： Fix the unit_test for adding new verify status

### CODE REVIEW

File: include/internal/libspdm_common_lib.h [hunk 1]
```diff
@@ -595,6 +595,42 @@ bool libspdm_generate_cert_chain_hash(libspdm_context_t *spdm_context,
...
+ * @param  cert_chain_buffer_size          size in bytes of the certificate chain buffer.
+ * @param  is_requester                   Indicates if it is a requester message.
+ *
+ * @retval true  Peer certificate chain buffer integrity verification passed.
+ * @retval false Peer certificate chain buffer integrity verification failed.
+ **/
+bool libspdm_verify_peer_cert_chain_buffer_integrity(libspdm_context_t *spdm_context,
+                                                     const void *cert_chain_buffer,
+                                                     size_t cert_chain_buffer_size,
+                                                     bool is_requester);
```
Comment (steven-bellock) [inline include/internal/libspdm_common_lib.h:613]: `is_requester` is not needed. Its value is stored in the `spdm_context`.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/internal/libspdm_common_lib.h (hunk 1 above)
Comment (Wenxing-hou) [reply include/internal/libspdm_common_lib.h]: I forgot we have added the `is_requester` in `spdm_context`. I have changed the code. Thanks.

File: library/spdm_common_lib/libspdm_com_crypto_service.c [hunk 2]
```diff
@@ -575,34 +575,26 @@ uint8_t libspdm_get_cert_slot_count(libspdm_context_t *spdm_context)
...
-    size_t root_cert_hash_size;
-    const uint8_t *received_root_cert;
-    size_t received_root_cert_size;
     bool result;
-    uint8_t root_cert_index;
     bool is_device_cert_model;
+    bool is_requester;
+
+    /*peer is_requester is opposite of local*/
+    is_requester = !spdm_context->local_context.is_requester;
```
Comment (steven-bellock) [inline library/spdm_common_lib/libspdm_com_crypto_service.c:597]: `is_requester` always specifies the local entity. This is backwards.

File: library/spdm_common_lib/libspdm_com_crypto_service.c [hunk 3]
```diff
@@ -710,6 +730,51 @@ bool libspdm_verify_peer_cert_chain_buffer(libspdm_context_t *spdm_context,
...
+ *
+ * @retval true  Peer certificate chain buffer verification passed.
+ * @retval false Peer certificate chain buffer verification failed.
+ **/
+bool libspdm_verify_peer_cert_chain_buffer(libspdm_context_t *spdm_context,
+                                           const void *cert_chain_buffer,
+                                           size_t cert_chain_buffer_size,
+                                           const void **trust_anchor,
+                                           size_t *trust_anchor_size,
+                                           bool is_requester)
```
Comment (steven-bellock) [inline library/spdm_common_lib/libspdm_com_crypto_service.c:751]: This isn't needed.

File: library/spdm_common_lib/libspdm_com_crypto_service.c [hunk 4]
```diff
@@ -710,6 +730,51 @@ bool libspdm_verify_peer_cert_chain_buffer(libspdm_context_t *spdm_context,
...
+{
+    bool result;
+
+    /*peer is_requester is opposite of local*/
+    if (is_requester == spdm_context->local_context.is_requester) {
```
Comment (steven-bellock) [inline library/spdm_common_lib/libspdm_com_crypto_service.c:756]: This isn't needed.

File: library/spdm_common_lib/libspdm_com_crypto_service.c (hunk 2 above)
Comment (Wenxing-hou) [reply library/spdm_common_lib/libspdm_com_crypto_service.c]: Thanks. I have changed the comment and made the `is_requeseter` to `is_peer_requester`.

File: library/spdm_common_lib/libspdm_com_crypto_service.c (hunk 3 above)
Comment (Wenxing-hou) [reply library/spdm_common_lib/libspdm_com_crypto_service.c]: Thanks. I have deleted the code.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**